data-prepper-api/src/main/java/org/opensearch/dataprepper/model/plugin/PluginConfigVariable.java

```diff
@@ -13,17 +13,29 @@
  * Interface for a Extension Plugin configuration variable.
  * It gives access to the details of a defined extension variable.
  *
- * @since 1.2
+ * @since 2.11
  */
 public interface PluginConfigVariable {
 
     /**
      * Returns the value of this variable.
+     *
+     * @return the value of this variable
      */
     Object getValue();
 
     /**
      * If this variable is updatable, this method helps to set a new value for this variable
+     *
+     * @param updatedValue         the new value to set
+     * @param secretVersionIdToSet new version id to set for the secret in AWS. This helps with idempotency
      */
-    void setValue(Object someValue);
+    void setValue(Object updatedValue, String secretVersionIdToSet);
```

Comment:
This information on the `secretVersionIdToSet` cannot be known by the caller because it binds the API to a single implementation.

So if you need to be tightly coupled to AWS Secrets, you may want to create a different structure for updating AWS Secrets.

Reply:
I agree we should not couple this class to secret specific argument